libnavui-voice/src/main/java/com/mapbox/navigation/ui/views/soundbutton/SoundButton.kt

```diff
@@ -0,0 +1,268 @@
+import android.util.AttributeSet
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.AlphaAnimation
+import android.view.animation.Animation
+import android.view.animation.AnimationSet
+import android.view.animation.DecelerateInterpolator
+import android.widget.TextView
+import androidx.annotation.StyleRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mapbox.navigation.ui.base.internal.utils.MultiOnClickListener
+import com.mapbox.navigation.ui.voice.R
+
+/**
+ * Sound Button view.
+ */
+class SoundButton : ConstraintLayout {
+
+    private lateinit var soundFab: FloatingActionButton
+    private lateinit var soundChipText: TextView
+    private var fadeInSlowOut: AnimationSet? = null
+    private var multiOnClickListener: MultiOnClickListener? = MultiOnClickListener()
+
+    private var primaryColor = 0
+    private var secondaryColor = 0
+
+    /**
+     * *true* if it's muted, otherwise *false*
+     */
+    var isMuted = false
+        private set
+
+    private companion object {
+        private const val ALPHA_VALUE_ZERO = 0f
+        private const val ALPHA_VALUE_ONE = 1f
+        private const val ANIMATION_DURATION_THREE_HUNDRED_MILLIS: Long = 300
+        private const val ANIMATION_DURATION_ONE_THOUSAND_MILLIS: Long = 1000
+    }
+
+    constructor(context: Context) : super(context) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        initAttributes(attrs)
+        init(context)
+    }
+
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int
+    ) : super(context, attrs, defStyleAttr) {
+        initAttributes(attrs)
+        init(context)
+    }
+
+    private fun initAttributes(attributeSet: AttributeSet?) {
+        val typedArray =
+            context.obtainStyledAttributes(attributeSet, R.styleable.MapboxStyleSoundButton)
+        primaryColor = ContextCompat.getColor(
+            context,
+            typedArray.getResourceId(
+                R.styleable.MapboxStyleSoundButton_soundButtonPrimaryColor,
+                R.color.mapbox_sound_button_primary
+            )
+        )
+        secondaryColor = ContextCompat.getColor(
+            context,
+            typedArray.getResourceId(
+                R.styleable.MapboxStyleSoundButton_soundButtonSecondaryColor,
+                R.color.mapbox_sound_button_secondary
+            )
+        )
+        typedArray.recycle()
+    }
+
+    private fun init(context: Context) {
+        inflate(context, R.layout.mapbox_button_sound, this)
+    }
+
+    /**
+     * Adds an onClickListener to the button
+     *
+     * @param onClickListener to add
+     */
+    override fun setOnClickListener(onClickListener: OnClickListener?) {
```

Comment:
`onClickListener: OnClickListener?` is nullable, but non-null in `removeOnClickListener `